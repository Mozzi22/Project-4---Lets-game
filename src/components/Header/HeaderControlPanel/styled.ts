import styled from 'styled-components';

export const StControl = styled.div`
        display: flex;
        justify-content: flex-end;
        align-items: center;
        height: 100%;
        width: 80%;
        margin-right: 20px;
        p { color: #fff }
`;
