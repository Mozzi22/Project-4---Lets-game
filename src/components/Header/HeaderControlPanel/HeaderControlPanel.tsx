import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import Button from 'src/components/UI/Button';
import Select from 'src/components/UI/Select';
import { Theme } from 'src/components/Hocs/withTheme';
import { options } from 'src/constants/ui';
import { StControl } from './styled';

const HeaderControlPanel = () => {
    const { i18n } = useTranslation();
    const { theme, changeTheme } = useContext(Theme);
    const [selected, setSelected] = React.useState('');

    React.useEffect(() => {
        const lastSelected = localStorage.getItem('lang');
        setSelected(lastSelected);
    }, [localStorage.getItem('lang')]);

    const onChangeLanguage = (lng, i18n) => {
        i18n.changeLanguage(lng);
        localStorage.setItem('lang', lng);
    };

    const handleChangeLanguage = (e) => {
        onChangeLanguage(e.target.value, i18n);
    };

    const toggleThemeMode = () => changeTheme(
        theme === 'dark' ? 'light' : 'dark',
    );

    return (
        <StControl>
            <Button
                id='theme-btn'
                fontSize='26px'
                width='40px'
                height='40px'
                margin='0 15px 0'
                borderRadius='0px'
                onClick={toggleThemeMode}
                bgColor='transparent'
                backgroundSize='cover'
                hoverColor='transparent'
                bgColorDisabled='transparent'
            />
            <Select
                id={options.id}
                options={options}
                onChange={handleChangeLanguage}
                value={selected || ''}
                width='70px'
                bgColor='#b16fdd'
            />
        </StControl>
    );
};

export default HeaderControlPanel;
