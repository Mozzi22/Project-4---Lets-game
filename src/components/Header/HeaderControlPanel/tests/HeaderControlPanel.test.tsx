import React from 'react';
import { shallow, mount } from 'enzyme';
import HeaderControlPanel from '../HeaderControlPanel';

jest.spyOn(localStorage, "setItem");
localStorage.setItem = jest.fn();

jest.spyOn(localStorage.__proto__, 'setItem');
localStorage.__proto__.setItem = jest.fn();

jest.mock('react-i18next', () => ({
   useTranslation: jest
     .fn()
     .mockReturnValue({ i18n: { changeLanguage: jest.fn() }, t: jest.fn() }),
 }));

describe('HeaderControlPanel', () => {
  it('Should match snapshot', () => {
    const component = shallow(<HeaderControlPanel/>);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have Button', () => {
    const component = mount(<HeaderControlPanel />);
    expect(component.find('Button')).toHaveLength(1);
  });
  it('saves the key to the storage', () => {
    const component = mount(<HeaderControlPanel />);
    component.find('Select').getElement().props.onChange({ target: { value: 'asad' } });
    expect(localStorage.setItem).toHaveBeenCalled();
  });
});
