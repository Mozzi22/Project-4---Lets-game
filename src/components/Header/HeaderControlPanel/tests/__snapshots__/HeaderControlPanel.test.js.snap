// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderControlPanel Should match snapshot 1`] = `"<div class=\\"styled__StControl-sc-875yq3-0 jmwBsH\\"><p></p> <button id=\\"theme_btn\\" type=\\"button\\" value=\\"light\\" color=\\"#fff\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-15f3n0p-0 isrQUZ\\"></button><button id=\\"theme_btn\\" type=\\"button\\" value=\\"dark\\" color=\\"#fff\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-15f3n0p-0 isrQUZ\\"></button><button id=\\"en\\" type=\\"button\\" value=\\"en\\" color=\\"#fff\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-15f3n0p-0 isrQUZ\\"></button><button id=\\"ru\\" type=\\"button\\" value=\\"ru\\" color=\\"#fff\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-15f3n0p-0 isrQUZ\\"></button></div>"`;
